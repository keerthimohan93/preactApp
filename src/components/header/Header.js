import { h } from 'preact';
import style from './style';

const Header = () => (
	<div>
		<header class={style.header}>
			<h1>My Collection</h1>
		</header>
	</div>
);
export default Header;
