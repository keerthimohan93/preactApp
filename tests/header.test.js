import { Provider } from 'preact-redux';
import { createStore } from 'redux';
import Header from '../src/components/header/Header';
import rootReducers from '../src/rootReducer';
// See: https://github.com/mzgoddard/preact-render-spy
import { shallow } from 'preact-render-spy';
const store = createStore(rootReducers);

describe('Initial Test of the Header', () => {
	test('Header renders correctly', () => {
		const context = shallow(<Header />);
		expect(context.find('h1').text()).toBe('My Collection');
	});
	test('it renders snapshot correctly', () => {
		const context = shallow(
			<Provider store={store}>
				<Header />
			</Provider>
		);
		expect(context).toMatchSnapshot();
	});
});
