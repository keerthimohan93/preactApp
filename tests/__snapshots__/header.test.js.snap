// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial Test of the Header it renders snapshot correctly 1`] = `
RenderContext {
  "0": T {
    "attributes": Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "children": Array [
      T {
        "attributes": undefined,
        "children": Array [],
        "key": undefined,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;
